### ISSUE TITLE
Global Fullscreen

### ISSUE THREAD

**Issue by ChristiaanDev** [hyprwm/Hyprland#1660]
Hello!

I have recently migrated from Sway to Hyprland and am missing one function that is holding me back from daily driving it.
In Sway there is a functionality which allows you to span a fullscreen window across several displays. This is known as "fullscreen global" and the functionality originated from i3wm.  

I have looked around in the documentation and have not been able to find something that allows us to replicate this behavior.  As far as I can tell this is not something we replicate with IPC either.

For reference, [here](https://github.com/swaywm/sway/pull/3423) is the relevant PR in Sway that added this functionality.

Thanks for all of the work so far!

**Comment by JustSimplyKyle** [comment]
You can *barely* script this basically by combining a fakefullscreen dispatcher and a resize dispatcher
But I agree, this can be a great addition

**Comment by IlyaMZP** [comment]
That feature would be very useful for flameshot. On sway I was able to use global fullscreen to make it span both monitors. On hyprland I have to make it floating and positioned at 0 0 on monitor 0, which is a lot of rules that may break if something changes.

**Comment by vaxerski** [comment]
if flameshot is a window, it's already doing it wrong.

**Comment by IlyaMZP** [comment]
> flameshot is a window, it's already doing it wrong

Yes, but they don't care about Wayland support whatsoever, it seems. Though flameshot is not the only thing that global fullscreen could be useful for.

**Comment by vaxerski** [comment]
> Yes, but they don't care about Wayland support whatsoever, it seems.

So why should I?

> Though flameshot is not the only thing that global fullscreen could be useful for.

Still, you can script (or even plugin) this

**Comment by IlyaMZP** [comment]
> So why should I?

Where did I say you should?
If you don't want to - don't bother.

**Comment by naythanN** [comment]
Hello, good end of year to everyone!
Has someone managed to make a global full screen work with Citrix Workspace?

Just to contextualize, I connect in a virtual desktop to work by using citrix. 

I tried scripting, I can get the window in the right resize and activate fakefullscreen, but it does not seem to work in this case apparently.

In other compositors, window managers, DEs. When the global fullscreen is active, I can use the two monitors as normal. (Two independent monitors that works normally with the host OS, no matter the OS). I tested in sway and works out of the box. Here, I can only use a single monitor. If I expand the window, it is like a big window. In the others, it is like two windows of the right size, each one with the windows from citrix.

If someone knows how to script this behavior I am willing to try as well.

Thanks!

**Comment by iMithrellas** [comment]
I would also really appreciate this functionality for freerdp /multimon usecase.

**Comment by RedKage** [comment]
Same, that's the only thing preventing me from moving away from i3.
RDP with multi monitor support is mandatory for me.

**Comment by iMithrellas** [comment]
> Same, that's the only thing preventing me from moving away from i3. RDP with multi monitor support is mandatory for me.

I have "solved" it for now by autostarting I3 with RDP in second TTY just for this use case, it's actually kinda nice, my remote machine is always one keystroke away.

**Comment by RedKage** [comment]
> > Same, that's the only thing preventing me from moving away from i3. RDP with multi monitor support is mandatory for me.
> 
> I have "solved" it for now by autostarting I3 with RDP in second TTY just for this use case, it's actually kinda nice, my remote machine is always one keystroke away.

Very smart.... thank you brother.
I will think about doing this

**Comment by teohz** [comment]
> Hello, good end of year to everyone! Has someone managed to make a global full screen work with Citrix Workspace?
> 
> Just to contextualize, I connect in a virtual desktop to work by using citrix.
> 
> I tried scripting, I can get the window in the right resize and activate fakefullscreen, but it does not seem to work in this case apparently.
> 
> In other compositors, window managers, DEs. When the global fullscreen is active, I can use the two monitors as normal. (Two independent monitors that works normally with the host OS, no matter the OS). I tested in sway and works out of the box. Here, I can only use a single monitor. If I expand the window, it is like a big window. In the others, it is like two windows of the right size, each one with the windows from citrix.
> 
> If someone knows how to script this behavior I am willing to try as well.
> 
> Thanks!

It's the same case for me - I also use Citrix for work, and unfortunately, I wasn't able to make multiple monitors work with Citrix as I hoped (I also came from sway and even after more than a year it's the only missing feature for me).

I may try @iMithrellas 's work-around as that may be the only option currently and looks promising.

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
